pkg/test/harness_test.go

```diff
@@ -81,3 +84,29 @@ func TestAddNodeCaches(t *testing.T) {
 	assert.Equal(t, "/var/lib/docker/data/kind-0", kindCfg.Nodes[0].ExtraMounts[0].HostPath)
 	assert.Equal(t, "/var/lib/docker/data/kind-1", kindCfg.Nodes[1].ExtraMounts[0].HostPath)
 }
+
+func TestRunBackgroundCommands(t *testing.T) {
+	h := Harness{
+		T: t,
+	}
+	h.TestSuite.StartControlPlane = true
+	commands := []v1beta1.Command{{
+		Command:    "sleep 1000000",
+		Background: true,
+	}}
+	h.TestSuite.Commands = commands
+
+	h.Setup()
+	defer h.Stop()
+
+	// setup creates bg processes
+	assert.Equal(t, 1, len(h.bgProcesses))
+	// process is alive
+	assert.NoError(t, h.bgProcesses[0].Process.Signal(syscall.Signal(0)))
+
+	// cleans up bg processes
+	h.Stop()
+	_, err := h.bgProcesses[0].Process.Wait()
```

Comment:
See my comment above about waiting

Reply:
this wait is to ensure we don't have flaky tests